models/userStatus.js

```diff
@@ -0,0 +1,121 @@
+  } catch (error) {
+    logger.error("Error in adding user status", error);
+    throw error;
+  }
+};
+
+/**
+ * @param userId {string} : id of the user
+ * @returns {Promise<userStatusModel|string>} : returns id of the deleted userStatus
+ */
+const deleteUserStatus = async (userId) => {
+  try {
+    const docs = await userStatusModel.where("userId", "==", userId).limit(1).get();
+    let docId;
+    if (docs._size > 0) {
+      docs.forEach((docData) => {
+        docId = docData.id;
+      });
+      await userStatusModel.doc(docId).delete();
+      return { userId };
+    }
+  } catch (error) {
+    logger.error(`error in deleting User Status Document . Reason - ${error}`);
+  }
+  return { message: `No UserStatus for ${userId} found.` };
+};
+
+/**
+ * @params userId {string} : id of the user
+ * @returns {Promise<userStatusModel|Object>} : returns the userStatus of a single user
+ */
+const getUserStaus = async (userId) => {
+  try {
+    const docs = await userStatusModel.where("userId", "==", userId).limit(1).get();
+    if (docs._size > 0) {
+      let data = {};
+      docs.forEach((docData) => {
+        data = docData.data();
+      });
+      return { userId, ...data };
+    }
+  } catch (error) {
+    logger.error(`error in fetching the User Status Document. Reason - ${error}`);
+  }
+  return { message: "User not found", userId, data: null };
+};
+
+/**
+ * @returns {Promise<userStatusModel|Array>} : returns an array of all the userStatus
+ */
+const getAllUserStaus = async (query) => {
+  const allUserStatus = [];
+  try {
+    let data;
+    if (!query.state) {
+      data = await userStatusModel.get();
+    } else {
+      data = await userStatusModel.where("currentStatus.state", "==", query.state).get();
+    }
+    data.forEach((doc) => {
+      const currentUserStatus = {
+        id: doc.id,
+        userId: doc.data().userId,
+        currentStatus: doc.data().currentStatus,
+        monthlyHours: doc.data().monthlyHours,
+      };
+      allUserStatus.push(currentUserStatus);
+    });
+  } catch (error) {
+    logger.error(`error in fetching the User Status of all Users. ${error}`);
+  }
+  return { allUserStatus };
+};
+
+/**
+ * @param userId { String }: Id of the User
+ * @param updatedData { Object }: Data to be Updated
+ * @returns Promise<userStatusModel|Object>
+ */
+
+const updateUserStatus = async (userId, updatedData) => {
+  try {
+    const docs = await userStatusModel.where("userId", "==", userId).limit(1).get();
+    if (docs._size > 0) {
+      let id;
+      docs.forEach((docData) => {
+        id = docData.id;
+      });
+      if (Object.keys(updatedData).includes("currentStatus")) {
+        const isUserOOO = updatedData.currentStatus.state === userState.OOO;
+        if (!isUserOOO) {
+          updatedData.currentStatus.until = "";
+        }
+        if (updatedData.currentStatus.state === userState.ACTIVE) {
+          updatedData.currentStatus.message = "";
+        }
```

Comment:
```js
        const updatedUserState = updatedData.currentStatus.state;
        const isUserOOO =  updatedUserState === userState.OOO;
        const isUserActive = updatedUserState === userState.ACTIVE;

        if (!isUserOOO) {
          updatedData.currentStatus.until = "";
        }
        if (isUserActive) {
          updatedData.currentStatus.message = "";
        }
```
If you think it's good. We can be consistent.

Reply:
Yeah @SanketDhabarde this is better and consistent. Thanks.